_posts/2021-06-23-#22261.md

```diff
@@ -83,9 +83,226 @@ commit:
+17:50 <glozow> ok the scenario is this: we have a transaction. the mempool has one with witness A and we call `BroadcastTransaction()` with the exact same tx, but different witness
+17:50 <lightlike> why would you malleate your own transactions?
+17:50 <glozow> idk 🤷 maybe there are other parties in it
+17:50 <glozow> it's a multisig, idk
+17:51 <michaelfolkson> lightlike: It is finding a cheaper witness after previously broadcasting a more expensive witness I think
+17:51 <glozow> maybe it has multiple spending paths and you/a counterparty both made transactions
+17:51 <LarryRuane> michaelfolkson cheaper meaning smaller (so lower fee)? 
+17:51 <michaelfolkson> It isn't deliberate malleation. And obviously the txid isn't malleated post SegWit
+17:52 <michaelfolkson> LarryRuane: Smaller, so yeah lower fee
+17:52 <glozow> this isn't a show-stopping bug, it's just not the behavior we'd expect
+17:53 <glozow> either way, we should try to relay the transaction in our mempool
+17:53 <michaelfolkson> "Prior to this commit, if BroadcastTransaction() is called with
+17:53 <michaelfolkson> relay=true, then it'll call RelayTransaction() using the txid/wtxid of
+17:53 <michaelfolkson> the new tx, not the txid/wtxid of the mempool tx."
+17:53 <michaelfolkson> And the mempool doesn't have the new transaction?
+17:53 <glozow> yes
+17:54 <michaelfolkson> Ok
+17:54 — michaelfolkson sweats
+17:54 <LarryRuane> what's the worst effect (at a high level) of not fixing these bugs? transactions not being relayed around the network?
+17:55 <glozow> good question. anybody have ideas?
+17:55 <michaelfolkson> 2) just means less effective relaying right? More failures
+17:55 <michaelfolkson> 1) is wasteful. Unnecessary relaying
+17:55 <lightlike> for the first one, it would seem the opposite: tx beinged inv'ed too much (although the network already knows about it)
+17:57 <glozow> right
+17:57 → kexkey joined (~kexkey@static-198-54-132-100.cust.tzulo.com)
+17:58 <glozow> no consensus failures. but worth fixing, ya?
+17:58 <lightlike> If i understand it correctly, "transactions not being relayed" would rather be a downside of the first fix: if we happen to have bad peers, broadcast to them, get a GETDATA but they don't relay any further, then we'd currently try again after ~24hours (when we have better peers), after the fix we wouldn't do this anymore.
+17:58 <LarryRuane> we're about out of time, so this would be too long of a discussion, but I notice there's no new or changed test code with this PR .. would be interesting to get into why, and generally, what kind of PRs need tests and what kind don't
+17:58 <michaelfolkson> LarryRuane: If you read the review comments tests were discussed
+17:59 <LarryRuane> ok thanks (hadn't done that, will do)
+17:59 <michaelfolkson> duncandean is working on tests it appears
+17:59 <michaelfolkson> I think everyone agrees it should have tests
+17:59 ⇐ kexkey_ quit (~kexkey@static-198-54-132-116.cust.tzulo.com): Ping timeout: 252 seconds
```

Comment:
You can configure your IRC client to not display the joins and quits.  We didn't usually add these to the meeting logs.